env0/resource_environment.go

```diff
@@ -78,13 +194,23 @@ func resourceEnvironmentRead(ctx context.Context, d *schema.ResourceData, meta i
 func resourceEnvironmentUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	apiClient := meta.(client.ApiClientInterface)
 
-	payload := getUpdatePayload(d)
+	if d.HasChanges("template_id", "revision", "configuration") {
```

Comment:
why do we need this part?
We have `ForceNew`, this one will means we create another resource on the same tf id
idk if we want that

Reply:
there is no force new on any of these fields
this is the deploy condition

Reply:
you can't change environment template id this will have to force new
you can change the revision or configuration

Reply:
e997b6f